handlers/instanceIdentityDocument.go

```diff
@@ -33,16 +33,16 @@ var (
 )
 
 func InstanceIdentityDocumentHandler(w http.ResponseWriter, r *http.Request) {
-
-	awsArn, err := util.ArnParse(metadata.Role)
+	// TODO: this was crashing because of a nil pointer dereference. Fix it!
+	awsArn, err := util.ArnParse("")
```

Comment:
Maybe we should bail on serving metadata if `metadata.Role` is not defined, and just serve a generic error?